File: 이것이 취업을 위한 코딩 테스트다/정렬/성적이_낮은_순서로_학생_출력하기/성적이_낮은_순서로_학생_출력하기.java
import java.util.*;

public class 성적이_낮은_순서로_학생_출력하기 {
  static class Student implements Comparable<Student>{
    public String name;
    public int score;
  
    Student(String name, int score) {
      this.name = name;
      this.score = score;
    }
  
    @Override
    public int compareTo(Student other) {
        if (this.score < other.score) {
          return - 1;
        }
  
        return 1;
    }
  }

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    ArrayList<Student> info = new ArrayList<>();

    for (int i = 0; i < n; i++) {
      String name = sc.next();
      int score = sc.nextInt();
      info.add(new Student(name, score));
    }

    Collections.sort(info);

    for(Student student : info) {
      System.out.print(student.name + " ");
    }
  }
}

File: 이것이 취업을 위한 코딩 테스트다/정렬/성적이_낮은_순서로_학생_출력하기/성적이_낮은_순서로_학생_출력하기.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

class Student {
  public:
    string name;
    int score;
    Student(string name, int score) {
      this->name = name;
      this->score = score;
    }

    bool operator <(Student &other) {
      return this-> score < other.score;
    }
};

int main() {
  int n;
  cin >> n;
  vector<Student>students;
  string name;
  int score;

  for(int i = 0; i < n; i++) {
    cin >> name >> score;
    students.push_back(Student(name, score));
  }

  sort(students.begin(), students.end());

  for(int i = 0; i < students.size(); i++) {
    cout << students[i].name << " ";
  }

  return 0;
}